src/peer-store/README.md

```diff
@@ -1,3 +1,156 @@
+
+**IMPORTANT:** This is one of the biggest design decisions to make (set vs events). The programmatic API is the easiest solution but it can provide users an API that they sould not need. If we go on an event-based approach, the `peerStore` should receive all the relevant subsystems (discovery, identifyService, ...) and sense the environment (via events) to gather the information that would need to be sent via the API. Thile the latest seems the best solution, it is the more complex one to implement, as we would ideally have an interface that those subsystems would need to implement and each time we have a new subsystem that needs to add data to the peerStore, we might need to update the `peer-store` codebase (or have a good set of abstractions).
+
+It is also important pointing out that users would be able to use `peerStore.protoBook.*`, so eventually we should move those into `peerStore._ protoBook.*` if we do not intend them to use it.
+
+---
+
+- `peerStore.get(peerId, options)`
+
+Get the information of a provided peer. The amount of information that we want can be customized with the following options, which are true by default:
+
+```js
+{
+  address: true,
+  proto: true,
+  key: true,
+  metadata: true
+}
+```
+
+- `peerStore.delete(peerId, [data])`
+
+Deletes the provided peer from every book. If data is provided, just remove the data from the books. The data should be provided as follows:
+
+```js
+{
+  address: [],
+  proto: [],
+  key: [],
+  metadata: []
+}
+```
+
+- `peerStore.peers(options)`
+
+Get an array of all the peers, as well as their information. The information intended can be customized with the following options, which are true by default:
+
+```js
+{
+  address: true,
+  proto: true,
+  key: true,
+  metadata: true
+}
+```
+
+## Address Book
+
+The `addressBook` keeps the known multiaddrs of a peer. The multiaddrs of each peer are not a constant and the Address book must have this into consideration.
+
+`Map<string, multiaddrInfo>`
+
+A `peerId.toString()` identifier mapping to a `multiaddrInfo` object, which should have the following structure:
+
+```js
+{
+  multiaddr: ,
+  validity: ,
+  confidence: 
+}
+```
+
+**Note:** except for multiaddr namings, the other properties are placeholders for now and might not be as described in the future milestones.
+
+- `addressBook.set()`
+- `addressBook.get()`
+- `getMultiaddrsForPeer()`
+- `addressBook.has()`
+- `addressBook.delete()`
+- `addressBook.peers()`
+
+It is important pointing out that the API methods which return arrays of data (`set`, `get`, `getMultiaddrsForPeer`) shuld return the `multiaddr` property of the `multiaddrInfo` and not the entire `multiaddrInfo` as the remaining data should be used internally. Should we consider having two datastructure instead?
```

Comment:
```suggestion
It is important pointing out that the API methods which return arrays of data (`set`, `get`, `getMultiaddrsForPeer`) should return the `multiaddr` property of the `multiaddrInfo` and not the entire `multiaddrInfo` as the remaining data should be used internally. Should we consider having two datastructure instead?
```

Reply:
Why should they only return the multiaddr? Is there a reason to not expose address confidence?

Reply:
hummm, I was thinking on the opposite, is there any reason to expose the address confidence?

What do you think on keeping `getMultiaddrsForPeer` where we would just return the `multiaddr` (with `/p2p/id`) and the `get` would return the complete object?

Reply:
If you wanted to manually prune addresses, confidence could be a way of going about this. This may be a small use case, but what I am mainly trying to get at is avoiding scenarios where we're "hiding" data. There shouldn't be a reason to not expose the data, however, the usability of `addressBook.get()` might make the most sense to just have it return multiaddrs for each of use and then a secondary method could eventually be added to get the other metadata.

> What do you think on keeping getMultiaddrsForPeer where we would just return the multiaddr (with /p2p/id) and the get would return the complete object?

I think this is a good work around for now, and avoids us breaking the current API the peerstore has.